src/test/timedata_tests.cpp

```diff
@@ -34,4 +38,58 @@ BOOST_AUTO_TEST_CASE(util_MedianFilter)
     BOOST_CHECK_EQUAL(filter.median(), 7);
 }
 
+void MultiAddTimeData(int n, int64_t offset)
+{
+    static int cnt = 0;
+    for (int i = 0; i < n; ++i) {
+        CNetAddr addr;
+        addr.SetInternal(std::to_string(++cnt));
+        AddTimeData(addr, offset);
+    }
+}
+
+
+BOOST_AUTO_TEST_CASE(addtimedata)
+{
+    BOOST_CHECK_EQUAL(GetTimeOffset(), 0);
+
+    //Part 1: Add large offsets to test a warning message that our clock may be wrong.
+    MultiAddTimeData(3, DEFAULT_MAX_TIME_ADJUSTMENT + 1); //filter size 4, initialized with 0
```

Comment:
You add 3, but filter size is 4, is vTimeOffsets seeded with a first sample somewhere in test setup ?

Reply:
Not in the test, but in `timedata.cpp`: vTimeOffsets is static and initialized with one element, offset 0:`static CMedianFilter<int64_t> vTimeOffsets(BITCOIN_TIMEDATA_MAX_SAMPLES, 0)`;
I later add the same amount of positive and negative elements so that this initial 0 element stays the median at the end. Will change the comment `initialized with 0` to a better wording.

Reply:
You're right, still have to get familiar with cpp multiple ways of initializing arrays.